bittensor/config.py

```diff
@@ -287,12 +287,22 @@ def __deepcopy__(self, memo) -> "config":
     def __repr__(self) -> str:
         return self.__str__()
 
+    def _remove_private_keys(self, d):
+        if "__parser" in d:
+            d.pop("__parser", None)
+        if "__is_set" in d:
+            d.pop("__is_set", None)
+        for k, v in list(d.items()):
+            if isinstance(v, dict):
+                self._remove_private_keys(v)
+        return d
+
     def __str__(self) -> str:
         # remove the parser and is_set map from the visible config
         visible = self.toDict()
         visible.pop("__parser", None)
-        visible.pop("__is_set", None)
-        return "\n" + yaml.dump(visible)
+        cleaned = self.remove_private_keys(visible)
```

Comment:
This removes `__parser` and `__is_set` from the config itself, would this cause issues in the long run? Or are they one-time parameters?

Reply:
Good to have this concern, but no need to worry! 

The `toDict()` method from the `DefaultMunch` object (which the `config` object is a subclass of) copies the content and returns them as a python dict, so we're actually not popping the state of the config object itself, but rather the copy of its content.